src/popup/components/Modals/AccountImport.vue

```diff
@@ -129,13 +129,18 @@ export default defineComponent({
       }
 
       discovering.value = true;
-      setMnemonic(mnemonicParsed);
-      setBackedUpSeed(true);
-      await discoverAccounts();
-      props.resolve();
-      router.push(loginTargetLocation.value);
-      openEnableSecureLoginModal();
-      discovering.value = false;
+      try {
+        await setMnemonicAndInitializePassword(mnemonicParsed);
```

Comment:
this looks little bit weird to use function with a name "...InitializePassword". On mobile we're not using password protection. Maybe we can split the logic of setting the mnemonic and initializing the security stuff. Eg.:

```
initializeAuth(); // This would start biometric or password prot depending on env
setMnemonic(mnemonicParsed);
```

The reason for this comment is also the fact that the body of this `try` block looks randomly ordered. Person that reads it would see that we set password protection and at the end we set up biometric login.

Reply:
The reason for this is that we need to ask user for password _before_ setting the mnemonic on web, but on mobile we want to show the biometric login _after_ the wallet was restored because on mobile authentication is not required to set the mnemonic